File: server/src/main/java/org/dan/jadalnia/app/festival/pojo/FestivalInfo.java
package org.dan.jadalnia.app.festival.pojo;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.experimental.FieldDefaults;
import lombok.experimental.Wither;
import org.dan.jadalnia.app.festival.menu.MenuItem;
import org.dan.jadalnia.app.festival.pojo.FestivalState;
import org.dan.jadalnia.app.festival.pojo.Fid;

import java.time.Instant;
import java.util.List;

@Getter
@Wither
@Builder
@RequiredArgsConstructor
@FieldDefaults(makeFinal = true)
public class FestivalInfo {
    Fid fid;
    String name;
    FestivalState state;
    List<MenuItem> menu;
    Instant opensAt;
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/festival/pojo/FestivalInfo.kt
package org.dan.jadalnia.app.festival.pojo

import org.dan.jadalnia.app.festival.menu.MenuItem
import java.time.Instant

data class FestivalInfo(
        val fid: Fid,
        val name: String,
        val state: FestivalState,
        val menu: List<MenuItem>,
        val params: FestParams,
        val opensAt: Instant) {

    fun withState(state1: FestivalState) = copy(state = state1)
    fun withMenu(menu: List<MenuItem>) = copy(menu = menu)
}
